tests/aws/services/opensearch/test_opensearch.py

```diff
@@ -246,6 +256,92 @@ def _search():
         test_user_client.create("new-index2", id="new-index-id2", body={})
         test_user_client.index(test_index_name, body={"test-key1": "test-value1"})
 
+    @markers.aws.validated
+    def test_sql_plugin(self, opensearch_create_domain, aws_client, snapshot, account_id):
+        master_user_auth = ("admin", "QWERTYuiop123!")
+        domain_name = f"sql-test-domain-{short_uid()}"
+        access_policy = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Principal": {"AWS": "*"},
+                    "Action": "es:*",
+                    "Resource": f"arn:aws:es:*:{account_id}:domain/{domain_name}/*",
+                }
+            ],
+        }
+
+        # create a domain that works on aws
+        opensearch_create_domain(
+            DomainName=domain_name,
+            EngineVersion=OPENSEARCH_DEFAULT_VERSION,
+            ClusterConfig=ClusterConfig(InstanceType="t3.small.search", InstanceCount=1),
+            EBSOptions=EBSOptions(EBSEnabled=True, VolumeType="gp2", VolumeSize=10),
+            AdvancedSecurityOptions=AdvancedSecurityOptionsInput(
+                Enabled=True,
+                InternalUserDatabaseEnabled=True,
+                MasterUserOptions=MasterUserOptions(
+                    MasterUserName=master_user_auth[0],
+                    MasterUserPassword=master_user_auth[1],
+                ),
+            ),
+            NodeToNodeEncryptionOptions=NodeToNodeEncryptionOptions(Enabled=True),
+            EncryptionAtRestOptions=EncryptionAtRestOptions(Enabled=True),
+            DomainEndpointOptions=DomainEndpointOptions(EnforceHTTPS=True),
+            AccessPolicies=json.dumps(access_policy),
+        )
+        endpoint = aws_client.opensearch.describe_domain(DomainName=domain_name)["DomainStatus"][
+            "Endpoint"
+        ]
+
+        # ensure opensearch-sql plugin is installed
+        plugins_url = f"https://{endpoint}/_cat/plugins?s=component&h=component"
+        response = requests.get(
+            plugins_url, auth=master_user_auth, headers={"Accept": "application/json"}
+        )
+        installed_plugins = {plugin["component"] for plugin in response.json()}
```

Comment:
I'm curious about the use of a set here. Does this endpoint return duplicate entries?